File: api/src/main/java/com/grash/model/LocationImage.java
package com.grash.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.Date;
import java.util.List;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "location_image")
public class LocationImage {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @NotNull
    @Column(nullable = false)
    private String name;
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "file_id")
    private File file;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "location_id", nullable = false)
    @NotNull
    @JsonIgnore
    private Location location;
    
    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "created_at", nullable = false, updatable = false)
    private Date createdAt;
    
    @UpdateTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "updated_at", nullable = false)
    private Date updatedAt;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by_id")
    private OwnUser createdBy;
    
    @Column(name = "is_active", nullable = false)
    private Boolean isActive = true;
    
    @OneToMany(mappedBy = "locationImage", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnore
    private List<AssetHotspot> hotspots;
}

File: frontend/src/models/owns/locationImage.ts
import { AssetHotspot } from './assetHotspot';

export default interface LocationImage {
  id: number;
  name: string;
  description?: string;
  fileId: number;
  fileUrl: string;
  locationId: number;
  locationName: string;
  createdAt: string;
  updatedAt: string;
  createdById?: number;
  createdByName?: string;
  isActive: boolean;
  hotspots?: AssetHotspot[];
}

export interface LocationImageMiniDTO {
  id: number;
  name: string;
  fileUrl: string;
  locationId: number;
}

export interface CreateLocationImageRequest {
  name: string;
  description?: string;
  locationId: number;
}
